Frameworks/AddressBook/ABRecord.mm

```diff
@@ -47,12 +83,117 @@ bool ABRecordSetValue(ABRecordRef record, ABPropertyID property, CFTypeRef value
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes Properties that return a Multi-Value are not yet supported, and a few properties
+        (alternate birthday, creation/modification dates, person kind, phonetic middle name)
+        don't have direct equivalents.
 */
 CFTypeRef ABRecordCopyValue(ABRecordRef record, ABPropertyID property) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (record == nullptr) {
+        return nullptr;
+    }
+
+    _ABContact* person = (__bridge _ABContact*)record;
+
+    if (property == kABPersonFirstNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.firstName;
+    } else if (property == kABPersonLastNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.lastName;
+    } else if (property == kABPersonMiddleNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.middleName;
+    } else if (property == kABPersonPrefixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNamePrefix;
+    } else if (property == kABPersonSuffixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNameSuffix;
+    } else if (property == kABPersonNicknameProperty) {
```

Comment:
can you break these down into various property sections and do a check & return.
The function itself it over 100 lines of code and the multitude of if statements makes it much more cumbersome.

Also if an 'if' statement return's the following can be an 'if' rather than an 'else if'